### PR TITLE
beefy: error logs for validators with dummy keys

### PR DESCRIPTION
**Pull request by acatangiu**
This outputs:
```
2024-04-02 14:36:02.135 ERROR tokio-runtime-worker beefy: 🥩 for session starting at block 21990151
no BEEFY authority key found in store, you must generate valid session keys
(https://wiki.polkadot.network/docs/maintain-guides-how-to-validate-polkadot#generating-the-session-keys)
```
error log entry, once every session, for nodes running with `Role::Authority` that have no public BEEFY key in their keystore

### CODE REVIEW

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5730359

**Comment by acatangiu** [comment]
Locally tested:

- authority with no BEEFY public key: prints error log
- authority with good BEEFY public key: doesn't print anything

don't know yet how to set up local run to run authority with dummy key to check it also prints error log

File: substrate/client/consensus/beefy/src/worker.rs [hunk 1]
```diff
@@ -348,9 +350,24 @@ impl<B: Block> PersistedState<B> {
...
+		if is_authority {
+			let mut beefy_key_present = false;
+			for key in key_store.public_keys().unwrap_or_default() {
+				let id_raw: &[u8] = key.as_slice();
+				if id_raw[0..4] != *b"beef" {
+					// Non-dummy key found.
+					beefy_key_present = true;
+				}
+			}
+			if !beefy_key_present {
```
Comment (bkchr) [inline substrate/client/consensus/beefy/src/worker.rs:363]:
```suggestion
			if key_store.public_keys().map_or(false, |k| k.is_empty()) {
```

Comment (bkchr) [reply substrate/client/consensus/beefy/src/worker.rs:363]: No one is adding these dummy keys to its keystore and even if they have done this, they have done this on purpose manually. 

So, we can just check if they have any key at all for BEEFY.

**Review by bkchr — APPROVED** [review]

File: substrate/client/consensus/beefy/src/worker.rs [hunk 2]
```diff
@@ -348,9 +350,24 @@ impl<B: Block> PersistedState<B> {
...
+		// verify we're not using BEEFY dummy key when role is authority.
+		if is_authority {
+			let mut beefy_key_present = false;
+			for key in key_store.public_keys().unwrap_or_default() {
+				let id_raw: &[u8] = key.as_slice();
+				if id_raw[0..4] != *b"beef" {
+					// Non-dummy key found.
+					beefy_key_present = true;
+				}
+			}
```
Comment (sandreim) [inline substrate/client/consensus/beefy/src/worker.rs:362]: Could be rewritten to smth like: 

```
 let beefy_key_present = key_store.public_keys().iter().any(|key| key.as_slice()[0..4] != *b"beef");
```

**Comment by acatangiu** [comment]
> don't know yet how to set up local run to run authority with dummy key to check it also prints error log

actually, dummy key is registered in runtime through `Session::upgrade_keys::<OldSessionKeys, _>(transform_session_keys);` (see https://github.com/polkadot-fellows/runtimes/pull/65) - which doesn't populate anything in offchain keystore, so it would just be empty (no BEEFY keys) unless validator operator explicitly created some

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5730912

**Review by svyatonik — APPROVED** [review]

**Pull request merged**